### PR TITLE
Revert backend api actions

### PR DESCRIPTION
**Pull request by escherize**
Removing the exploratory work around api actions from master.

This is a straight up revert of all the backend api/actions commits, except for these commits:

1) I deleted the api docs for actions [here](https://github.com/metabase/metabase/pull/22755/commits/58c4bd8c27c8c0875e2f7c285fd1433ddeaa3707)
2) put back in some docstring changes that help in master [here](https://github.com/metabase/metabase/pull/22755/commits/f5c4af1e80aa8984e67eb1d6c227d4f82361b946)
3) remove this [precondition](https://github.com/metabase/metabase/pull/22755/commits/9e23f6f1bb49ab4196f57026c4c1603b37d1d4b6) that was causing [these sorts of test failures](https://app.circleci.com/pipelines/github/metabase/metabase/36335/workflows/7f478a45-8c59-4e74-ba44-5114cbd9fae8/jobs/1901873/tests#failed-test-0)

### CODE REVIEW

**Comment by escherize** [comment]
Should https://github.com/metabase/metabase/pull/22755/files#diff-1fb7cd6dc14f96149b2b775f598268d8053ee68dca276c40001b41ca4965cff6L910-L914 stay in? it doesn't feel action related to me

**Comment by jeff-bruemmer** [comment]
We should have someone from the backend team approve the PR before merging.

For the docs portion, just FYI: the API docs are autogenerated by running `clojure -M:ee:run api-documentation`, which will update/overwrite the existing API docs. We only publish docs from the release branch on our website, but running that script in master will keep the docs files up to date in the repo. Alternatively, since you're only deleting that one markdown file, you could just delete the link to the /api/actions.md page from the API index page. 

None of this is a blocker for this PR, as we'll run the script at some point down the line which will remove that link. So you're good to go from my end.

**Comment by escherize** [comment]
Thank you for your explaination @jeff-bruemmer - I saw that you were the author on that markdown file. and wanted to make sure that that piece was good.

I'll remove my request for your review now. 🙏

**Comment by camsaul** [comment]
That's weird I'm 90% sure I removed that precondition on the branch I added it before it was even reviewed

File: test/metabase/test/data/h2.clj [hunk 1]
```diff
@@ -51,7 +50,7 @@
   (defmethod sql.tx/field-base-type->sql-type [:h2 base-type] [_ _] database-type))
 
 (defmethod tx/dbdef->connection-details :h2
-  [_driver context dbdef]
+  [_ context dbdef]
```
Comment (camsaul) [inline test/metabase/test/data/h2.clj:53]: Some of the stuff like this is just a general improvement and we don't need to revert it

File: test/metabase/test/data/impl.clj [hunk 2]
```diff
@@ -295,7 +295,7 @@
   (copy-db-fks! old-db-id new-db-id))
 
 (def ^:dynamic *db-is-temp-copy?*
-  "Whether the current test database is a temp copy created with the [[metabase.test/with-temp-copy-of-db]] macro."
+  "Whether the current test database is a temp copy created with the `with-temp-copy-of-db` macro."
```
Comment (camsaul) [inline test/metabase/test/data/impl.clj:298]: Same -- don't need to revert this

File: test/metabase/test/data/interface.clj [hunk 3]
```diff
@@ -464,7 +464,7 @@
 (p.types/deftype+ ^:private EDNDatasetDefinition [dataset-name def]
   pretty/PrettyPrintable
   (pretty [_]
-    (list `edn-dataset-definition dataset-name)))
+    (list 'edn-dataset-definition dataset-name)))
```
Comment (camsaul) [inline test/metabase/test/data/interface.clj:467]: Same -- don't need to revert this

File: test/metabase/test/data/interface.clj [hunk 4]
```diff
@@ -496,12 +496,13 @@
 (p.types/deftype+ ^:private TransformedDatasetDefinition [new-name wrapped-definition def]
   pretty/PrettyPrintable
   (pretty [_]
-    (list `transformed-dataset-definition new-name (pretty/pretty wrapped-definition))))
+    (list 'transformed-dataset-definition new-name (pretty/pretty wrapped-definition))))
```
Comment (camsaul) [inline test/metabase/test/data/interface.clj:499]: Same -- don't need to revert this

File: test/metabase/test/data/postgres.clj [hunk 5]
```diff
@@ -72,7 +72,7 @@
   ;; add an additional statement to the front to kill open connections to the DB before dropping
   (cons
    (kill-connections-to-db-sql database-name)
-   (apply (get-method ddl/drop-db-ddl-statements :sql-jdbc/test-extensions) driver dbdef options)))
+   (apply (get-method ddl/drop-db-ddl-statements :sql-jdbc/test-extensions) :postgres dbdef options)))
```
Comment (camsaul) [inline test/metabase/test/data/postgres.clj:75]: This was a bugfix and we don't need to revert it

**Review by camsaul — APPROVED** [review]

**Pull request merged**